aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -0,0 +1,411 @@
+  auto storage_offset = storage_offset_.value_or(self.storage_offset());
+  setStrided(result, size, stride, storage_offset);
+
+  // 0 sizes won't result in any change in the shape of the Tensor so we can
+  // skip it. Also if the memory is contiguous we don't need to do
+  // gather-scatter operations using graph.
+  if (size.size() > 0 && (!result.is_contiguous())) {
+
+    // If self itself was a view tensor, that means we need to chain the graphs
+    // else we will create a new entry in the cache
+    struct CachedGraph : public MPSCachedGraph
+    {
+      CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+      MPSGraphTensor* inputTensor_ = nil;
+      MPSGraphTensor* outputTensor_ = nil;
+      IntArrayRef size_;
+      IntArrayRef stride_;
+      int64_t storage_offset_;
+    };
+
+    MPSGraphCache* cache_ = MPSGraphCache::getInstance();
+
+    @autoreleasepool {
+      string lookup_key = mps::getStridedKey(self, self.sizes(), self.strides(),
+                      self.storage_offset());
+#if _DEBUG
+      std::cout << "Lookup key " << lookup_key << std::endl;
+#endif
+      CachedGraph* cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(lookup_key));
+
+      if(!cachedGraph) {
+        string insert_key = mps::getStridedKey(self,size, stride, storage_offset);
+#if _DEBUG
+        std::cout << "Insert key " << insert_key << std::endl;
+#endif
+        CachedGraph* insertCachedGraph = static_cast<CachedGraph *>(cache_->LookUp(insert_key));
+        if (!insertCachedGraph) {
+          MPSCachedGraph *tmpCachedGraph = cache_->CreateCachedGraph(insert_key, ^ MPSCachedGraph * () {
+            CachedGraph *newCachedGraph = nil;
+            @autoreleasepool {
+                MPSGraph* mpsGraph = make_mps_graph();
+                newCachedGraph = new CachedGraph(mpsGraph);
+
+                // Self is the input tensor we are creating view of
+                MPSGraphTensor* inputTensor = [mpsGraph placeholderWithShape : getMPSShape(self)
+                                                                    dataType : getMPSDataType(self.scalar_type())
+                                                                    name : nil];
+                newCachedGraph->inputTensor_ = inputTensor;
+                newCachedGraph->outputTensor_ = chainViewOperation(mpsGraph, size,
+                                                                   stride,
+                                                                   storage_offset,
+                                                                   inputTensor,
+                                                                   self);
+                newCachedGraph->size_ = size;
+                newCachedGraph->stride_ = stride;
+                newCachedGraph->storage_offset_ = storage_offset;
+            }
+            return newCachedGraph;
+          });
+          cachedGraph = static_cast<CachedGraph *>(tmpCachedGraph);
```

Comment:
I'm a little confused, `cachedGraph` looks dead at this point, is it doing anything?